packages/transition-common/src/services/schedules/Schedule.ts

```diff
@@ -54,11 +77,892 @@ export interface ScheduleAttributes extends GenericAttributes {
+    outboundIntervalSeconds: number;
+    inboundIntervalSeconds: number;
+    outboundTotalTimeSeconds: number;
+    inboundTotalTimeSeconds: number;
+    units: TransitUnit[];
+    outboundPath: TransitPath;
+    inboundPath?: TransitPath;
+    period?: any;
+}
+
+// Interface for calculateResourceRequirements options
+interface CalculateResourcesOptions {
+    period: any;
+    startAtSecondsSinceMidnight: number;
+    endAtSecondsSinceMidnight: number;
+    outboundTotalTimeSeconds: number;
+    inboundTotalTimeSeconds: number;
+    secondAllowed?: boolean;
+}
+
+export interface GenerateTripsWithIntervalsOptions {
+    startAtSecondsSinceMidnight: number;
+    endAtSecondsSinceMidnight: number;
+    outboundIntervalSeconds: number;
+    inboundIntervalSeconds: number;
+    outboundTotalTimeSeconds: number;
+    inboundTotalTimeSeconds: number;
+    units: TransitUnit[];
+    outboundPath: TransitPath;
+    inboundPath?: TransitPath;
+}
+
+// Interface for initializeUnits
+export interface InitializeUnitsOptions {
+    units: TransitUnit[];
+    startFromDestination: boolean;
+    startTime: number;
+}
+
+// Interface for generateDepartureSchedules
+export interface GenerateDepartureSchedulesOptions {
+    startTime: number;
+    endTime: number;
+    outboundIntervalSeconds: number;
+    inboundIntervalSeconds: number;
+    outboundTotalTimeSeconds: number;
+    inboundTotalTimeSeconds: number;
+    startFromDestination: boolean;
+    hasInboundPath: boolean;
+}
+
+// Interface for processing scheduled departures
+export interface ProcessDeparturesOptions {
+    currentTime: number;
+    nextOutbound: number;
+    nextInbound: number;
+    units: TransitUnit[];
+    outboundPath: TransitPath;
+    inboundPath?: TransitPath;
+    outboundTotalTimeSeconds: number;
+    inboundTotalTimeSeconds: number;
+    trips: any[];
+    usedUnitsIds: Set<number>;
+    outboundDepartures: number[];
+    inboundDepartures: number[];
+}
+
+// Interface for processDeparture
+export interface ProcessDepartureOptions {
+    currentTime: number;
+    totalTimeSeconds: number;
+    units: TransitUnit[];
+    path: TransitPath;
+    trips: any[];
+    direction: UnitDirection;
+}
+
+// Interface for generateTrip
+export interface GenerateTripOptions {
+    tripStartAtSeconds: number;
+    unit: TransitUnit;
+    path: TransitPath;
+    segments: any[];
+    nodes: string[];
+    dwellTimes: number[];
+    blockId?: string | null;
+}
+
+// interface for schedule generation strategies
+export interface ScheduleGenerationStrategy {
+    calculateResourceRequirements(options: CalculateResourcesOptions): {
+        units: TransitUnit[];
+        outboundIntervalSeconds: number;
+        inboundIntervalSeconds: number;
+    };
+
+    generateTrips(options: GenerateTripsOptions): {
+        trips: SchedulePeriodTrip[];
+        realUnitCount: number;
+    };
+}
+export abstract class BaseScheduleStrategy implements ScheduleGenerationStrategy {
```

Comment:
More of a note to ourselves, no need to change it. 
Should all of this be in transition-backend instead of -common ? Don't think this part should be running in the frontend.